### ISSUE TITLE
Is there a way to blacklist a strategy for a particular runner?

### ISSUE THREAD

**Issue by dbarnett** [janko/vim-test#434]
If I have a runner that's not compatible with certain strategies, can I somehow configure it to fall back to another strategy (either in the runner implementation or as a user in my vimrc)?

Motivation: The vroom tool (#433) invokes vim under the hood and currently needs to run inside a real terminal. If I try to set up a runner for it and invoke it using the dispatch strategy that I prefer for other runners, the tool just blows up saying input/output is not a terminal.

**Comment by codeinabox** [comment]
Good question, I am not aware of such functionality but @janko is the best to comment on this. Aside from vroom are there any other runners that have this requirement?

**Comment by dbarnett** [comment]
As a hard requirement, I doubt it. Vroom just inherits that limitation from needing to run vim in-process. I could imagine users wanting to special case though, using more of a full terminal for a more verbose tool and something lighter for others.

**Comment by dbarnett** [comment]
Actually, let's focus on the user override case: Would it be okay to add an option for a `b:test_strategy` override so users could force a certain strategy for a certain buffer? Then users could hack around compatibility problems in a pinch, and that offers all kinds of flexibility to users for adding logic to handle corner cases.

Beyond that, I filed #436 for avoiding incompatible strategies in the vroom runner in particular. I realized the behavior I'd want for that is even a little more nuanced than just blacklisting "dispatch" and that issue seems pretty specific to just the vroom runner.

**Comment by codeinabox** [comment]
Was about to comment on #436 but I think it's best to keep this the main thread. A couple of thoughts and questions:

- How would the list of incompatible strategies be maintained and who would maintain it?
- Given that new strategies can be added would it make sense to have a whitelist instead?
- What would the user experience be if a user hasn't selected a compatible strategy?
- Is Vroom the only runner with this requirement? If so I'd be inclined to just list the compatible strategies in the documentation and README instead of making changes to the vim-test core

**Comment by dbarnett** [comment]
Yep, behavior does get a little complicated if it's a core feature and I do suspect it's going to be just Vroom running into compatibility issues, so I was leaning towards just
  (a) documenting the incompatibility, and
  (b) providing a way for users to pick strategy more case-by-case (see `b:test_strategy` suggestion)

**Comment by codeinabox** [comment]
Regarding your last point do you mean like this which vim-test already does?

```vim
let test#strategy = {
  \ 'nearest': 'neovim',
  \ 'file':    'dispatch',
  \ 'suite':   'basic',
\}
```

**Comment by dbarnett** [comment]
That's not per-runner, but similar yes. I'd either need a user override to say "in this buffer, use this set of strategies instead" or a separate mapping to say "when running tests with this runner/filetype, use this strategy".
